java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -2525,6 +2526,180 @@ public void testAnnotatedTestMethodsArrayExpressionControl() throws IOException
+            "import java.lang.annotation.Target;",
+            "@Repeatable(NewToggleTreatedSet.class)",
+            "@Target({ElementType.METHOD})",
+            "@interface NewToggleTreated {",
+            "  String[] value();",
+            "}")
+        .addOutputLines(
+            "NewToggleTreated.java",
+            "package com.uber.piranha;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Repeatable;",
+            "import java.lang.annotation.Target;",
+            "@Repeatable(NewToggleTreatedSet.class)",
+            "@Target({ElementType.METHOD})",
+            "@interface NewToggleTreated {",
+            "  String[] value();",
+            "}")
+        .addInputLines(
+            "NewToggleControl.java",
+            "package com.uber.piranha;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Repeatable;",
+            "import java.lang.annotation.Target;",
+            "@Repeatable(NewToggleControlSet.class)",
+            "@Target({ElementType.METHOD})",
+            "@interface NewToggleControl {",
+            "  String[] value();",
+            "}")
+        .addOutputLines(
+            "NewToggleControl.java",
+            "package com.uber.piranha;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Repeatable;",
+            "import java.lang.annotation.Target;",
+            "@Repeatable(NewToggleControlSet.class)",
+            "@Target({ElementType.METHOD})",
+            "@interface NewToggleControl {",
+            "  String[] value();",
+            "}");
+  }
+
+  @Test
+  public void testAnnotatedTestMethodsWithCustomSpecTreated() throws IOException {
+
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", "STALE_FLAG");
+    b.putFlag("Piranha:IsTreated", "true");
+    b.putFlag("Piranha:Config", "config/properties.json");
+
+    BugCheckerRefactoringTestHelper bcr =
+        BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+    bcr = bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+
+    bcr = addHelperClasses(bcr);
+    bcr =
+        addToggleTreatedAndControlAnnotationHelperClasses(
+            bcr); // Adds @NewToggleTreated and @NewToggleControl
+    bcr.addInputLines(
+            "TestClass.java",
+            "package com.uber.piranha;",
+            "class TestClass {",
+            "  public void foo () {}",
+            "  @NewToggleTreated(\"STALE_FLAG\")",
+            "  public void x () {}",
+            "  @NewToggleControl(\"STALE_FLAG\")",
+            "  public void y () {}",
+            "  @NewToggleTreated(\"STALE_FLAG\")",
+            "  @NewToggleTreated(\"OTHER_FLAG_1\")",
+            "  public void z () {}",
+            "  @NewToggleControl(\"STALE_FLAG\")",
+            "  @NewToggleTreated(\"OTHER_FLAG_1\")",
```

Comment:
can we also add a function with one annotation for `OTHER_FLAG_1` and ensure that there are no changes?